src/cata_utility.cpp

```diff
@@ -509,3 +509,47 @@ bool read_from_file_optional( const std::string &path, JsonDeserializer &reader
         reader.deserialize( jsin );
     } );
 }
+
+std::string native_to_utf8( const std::string &str )
+{
+    if( !get_options().has_option( "ENCODING_CONV" ) || !get_option<bool>( "ENCODING_CONV" ) ) {
+        return str;
+    }
+#if defined(_WIN32) || defined(WINDOWS)
+    // native encoded string --> Unicode sequence --> UTF-8 string
+    int unicode_size = MultiByteToWideChar( CP_ACP, 0, str.c_str(), -1, NULL, NULL ) + 1;
```

Comment:
Naturally, this compiled fine when the PR builder hit it (for speed reasons, it only builds lincurses), but none of the Windows compilers are actually happy with this:

```
src/cata_utility.cpp: In function 'std::string native_to_utf8(const string&)':
src/cata_utility.cpp:497:83: error: passing NULL to non-pointer argument 6 of 'int MultiByteToWideChar(UINT, DWORD, LPCCH, int, LPWSTR, int)' [-Werror=conversion-null]
     int unicode_size = MultiByteToWideChar( CP_ACP, 0, str.c_str(), -1, NULL, NULL ) + 1;
                                                                                   ^
src/cata_utility.cpp: In function 'std::string utf8_to_native(const string&)':
src/cata_utility.cpp:519:87: error: passing NULL to non-pointer argument 6 of 'int MultiByteToWideChar(UINT, DWORD, LPCCH, int, LPWSTR, int)' [-Werror=conversion-null]
     int unicode_size = MultiByteToWideChar( CP_UTF8, 0, str.c_str(), -1, NULL, NULL ) + 1;
                                                                                       ^
```

<http://ci.narc.ro/job/Cataclysm-Matrix/Graphics=Curses,Platform=Windows/6218/console>